### PR TITLE
Various fixes and additions pulled from #424

### PR DESCRIPTION
**Pull request by martijnbastiaan**
I'm trying to get #424 in a mergeable state, but the branch has accumulated a whole bunch of patches. I'd like to get these in separately to reduce cognitive load when rebasing / moving stuff around.

### CODE REVIEW

File: bittide/src/Bittide/Wishbone.hs [hunk 1]
```diff
@@ -103,6 +107,13 @@ ilaWb ::
     (Wishbone dom 'Standard addrW a)
 ilaWb = Circuit $ \(m2s, s2m) ->
   let
```
Comment (lmbollen) [inline bittide/src/Bittide/Wishbone.hs:109]: I´d consider enabling the user to change the `stages` and `depth` field of `IlaConfig`.
Either by passing those as argument or allowing `IlaConfig` as argument and adjusting the fields internally. (with a check ofc)

File: nix/bin/install-openocd-vexriscv-udev [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  exec_print sudo usermod -aG "${DEV_GROUP}" "${CURRENT_USER}"
+  exec_print newgrp "${DEV_GROUP}"
+fi
+
+if [ -f "${UDEV_RULES_D}/${UDEV_RULES_BASENAME}" ]; then
+  echo "'${UDEV_RULES_D}/${UDEV_RULES_BASENAME}' already exists, doing nothing"
+else
+  exec_print sudo cp "$OPENOCD_DIST/share/openocd/contrib/${UDEV_RULES_BASENAME}" "${UDEV_RULES_D}"
+  exec_print sudo udevadm control --reload-rules
+fi
```
Comment (lmbollen) [inline nix/bin/install-openocd-vexriscv-udev:46]: I'm conflicted whether this belongs in this PR.

**Review by lmbollen — APPROVED** [review]
LGTM modulo a comment and a question, let's discuss offline.

File: nix/bin/install-openocd-vexriscv-udev (hunk 2 above)
Comment (martijnbastiaan) [reply nix/bin/install-openocd-vexriscv-udev:46]: Discussed this offline: while this does belong in another PR, I'm feeling a bit overwhelmed by all the changes there, so I'd like to merge it separately.

File: bittide/src/Bittide/Wishbone.hs (hunk 1 above)
Comment (martijnbastiaan) [reply bittide/src/Bittide/Wishbone.hs:115]: Done, does this look good to you?

**Pull request merged**